### ISSUE TITLE
Action Popover - Gap between focus outline and download option fill

### ISSUE THREAD

**Issue by sianford** [Sage/carbon#5998]
### Current behaviour

There is a gap between the top and bottom of the focus outline and the fill boundary of the Download item in the Popover list in the `action-popover--action-popover-component-download-button` story.

<img width="200" alt="Action_Popover_with_download_button_focus_outline" src="https://user-images.githubusercontent.com/12828099/234873739-9caa6153-8b8d-4a7b-8cef-53b0a91d5d20.png">

### Expected behaviour

There should be no gap between the element and the focus outline.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/action-popover--action-popover-component-download-button

### JIRA Ticket (Sage Only)

FE-5774

### Suggested Solution

Bring the focus outline in to the element fill boundary.

### Carbon Version

118.1.0

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by Parsium** [comment]
FE-5774
